cli/hooks/webpack.js

```diff
@@ -0,0 +1,129 @@
+// eslint-disable-next-line security/detect-child-process
+const { execSync } = require('child_process');
+const fs = require('fs');
+const path = require('path');
+const semver = require('semver');
+const util = require('util');
+require('colors');
+
+const WebpackService = require('../lib/webpack/service');
+
+const MIN_APPCD_VERSION = '3.1.0';
+let AppcdClient;
+
+exports.id = 'ti.webpack';
+exports.init = (logger, config, cli) => {
+	let isBuildCommand = false;
+	let isWebpackProject = false;
+
+	cli.on('cli:command-loaded', (hookData) => {
+		const command = hookData.command;
+		if (command.name !== 'build') {
+			return;
+		}
+
+		isBuildCommand = true;
+	});
+
+	cli.on('cli:post-validate', () => {
+		if (!isBuildCommand) {
+			return;
+		}
+
+		isWebpackProject = !!cli.tiapp.webpack;
+		if (!isWebpackProject) {
+			return;
+		}
+
+		let appcdRootPath;
+		if (process.env.APPC_ENV) {
+			// we were invoked by appc-cli, load bundled daemon client.
+			appcdRootPath = resolveModuleRoot('appcd');
+			if (!fs.existsSync(appcdRootPath)) {
+				throw new Error('Unable to find Appcelerator Daemon inside the appc-cli. Please make sure to use a recent appc-cli version. You can install/select it with  "appc use latest".');
+			}
+			ensureAppcdVersion(appcdRootPath, MIN_APPCD_VERSION, true);
+		} else {
+			// plain ti-cli, load daemon from global modules
+			const globalModulesPath = execSync('npm root -g').toString().replace(/\s+$/, '');
```

Comment:
this seems potentially problematic on windows...

Reply:
`execSync` starts a new shell on windows and directly passes the command to that shell, so this shouldn't be an issue. I tested this on my Window 10 machine and works, but i'll ask QE to double check.